source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -734,11 +734,15 @@ The event API here is assumed to be like the standard `Python Event
                 # Initiate streaming hello or legacy hello
                 if connectTimeoutMS != 0:
                     set connection timeout to connectTimeoutMS+heartbeatFrequencyMS
-                response = call {helloCommand: 1, topologyVersion: previousDescription.topologyVersion, maxAwaitTimeMS: heartbeatFrequencyMS}
+                response = call {helloCommand: 1, helloOk: True, topologyVersion: previousDescription.topologyVersion, maxAwaitTimeMS: heartbeatFrequencyMS}
             else:
                 # The server does not support topologyVersion.
-                response = call {helloCommand: 1}
+                response = call {helloCommand: 1, helloOk: True}
 
+            # if server supports hello, then response.helloOk will be true
+            # and hello will be used for subsequent monitoring commands.
+            # if server does not support hello, then response.helloOk will be undefined
+            # and legacy hello will be used for subsequent monitoring commands.
             return ServerDescription(response, rtt=rttMonitor.average())
         except Exception as exc:
```

Comment:
To skip storing on the ServerDescription, reset `helloOk`:
```suggestion
        except Exception as exc:
        helloOk = versionedApi != null
```

Reply:
I added this logic to `setupConnection` so we would only have to do it in one place.